src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -655,13 +656,88 @@ protected override void PrepareContainerForItemOverride(DependencyObject element
 			}
 		}
 
+		private const string DragItemsFormatId = DataPackage.UnoPrivateDataPrefix + "__dragged__items__";
+
 		internal override void ContainerPreparedForItem(object item, SelectorItem itemContainer, int itemIndex)
 		{
 			base.ContainerPreparedForItem(item, itemContainer, itemIndex);
 
+			if (CanDragItems)
+			{
+				itemContainer.CanDrag = true;
+				itemContainer.DragStarting += OnItemContainerDragStarting;
+				itemContainer.DropCompleted += OnItemContainerDragCompleted;
+			}
+
 			ContainerContentChanging?.Invoke(this, new ContainerContentChangingEventArgs(item, itemContainer, itemIndex));
 		}
 
+		internal override void ContainerClearedForItem(object item, SelectorItem itemContainer)
+		{
+			itemContainer.DragStarting -= OnItemContainerDragStarting;
+			itemContainer.DropCompleted -= OnItemContainerDragCompleted;
+
+			base.ContainerClearedForItem(item, itemContainer);
+		}
+
+		public event DragItemsStartingEventHandler DragItemsStarting;
+		public event TypedEventHandler<ListViewBase, DragItemsCompletedEventArgs> DragItemsCompleted;
+
+		public static DependencyProperty CanReorderItemsProperty { get; } = DependencyProperty.Register(
+			nameof(CanReorderItems),
+			typeof(bool),
+			typeof(ListViewBase),
+			new FrameworkPropertyMetadata(default(bool)));
+
+		public bool CanReorderItems
+		{
+			get => (bool)GetValue(CanReorderItemsProperty);
+			set => SetValue(CanReorderItemsProperty, value);
+		}
+
+		public static DependencyProperty CanDragItemsProperty { get; } = DependencyProperty.Register(
+			nameof(CanDragItems),
+			typeof(bool),
+			typeof(ListViewBase),
+			new FrameworkPropertyMetadata(default(bool)));
```

Comment:
If this property changes, should drag subscriptions be added to current items? (Or removed but this seems less pressing, since you check listView.CanDragItems in the handler)